buffer/src/test/java/io/netty/buffer/AbstractByteBufTest.java

```diff
@@ -2894,6 +2897,89 @@ private void testGetReadOnlyDst(boolean direct) {
         assertEquals(0, readOnlyDst.position());
     }
 
+    @Test
+    public void testReadBytesAndWriteBytesWithFileChannel() throws IOException {
+        File file = File.createTempFile("file-channel", ".tmp");
+        RandomAccessFile randomAccessFile = null;
+        try {
+            randomAccessFile = new RandomAccessFile(file, "rw");
+            FileChannel channel = randomAccessFile.getChannel();
+            // channelPosition should never be changed
+            long channelPosition = channel.position();
+
+            byte[] bytes = {'a', 'b', 'c', 'd'};
+            int len = bytes.length;
+            ByteBuf buffer = releaseLater(newBuffer(len));
+            buffer.resetReaderIndex();
+            buffer.resetWriterIndex();
```

Comment:
This is for `SlicedByteBuf` because it doesn't set `writerIndex` to 0 by default.